apriltag_ros/src/continuous_detector.cpp

```diff
@@ -86,6 +97,81 @@ bool ContinuousDetector::refreshParamsCallback(std_srvs::Empty::Request& req,
   return true;
 }
 
+void ContinuousDetector::publishEmptyDetection(const std_msgs::Header& header)
+{
+  AprilTagDetectionArray tag_detection_array;
+  tag_detection_array.header = header;
+  tag_detections_publisher_.publish(tag_detection_array);
+}
+
+bool ContinuousDetector::isTagInFOV(const tf2::Transform& tag_pose) const
+{
+  if (!camera_model_.initialized())
+  {
+    ROS_WARN_STREAM_THROTTLE(1.0, "Camera not initialized, skipping FOV check");
+    return false;
+  }
+  if (tag_pose.getOrigin().getZ() <= 0)
+  {
+    // handles the case where Z is equal to zero (shouldn't happen) or less than zero (behind the camera)
+    return false;
+  }
+
+  // convert the 3d pose in the image frame to a 2d pixel frame
+  const auto image_pt = camera_model_.project3dToPixel(
+    cv::Point3f(
+      tag_pose.getOrigin().getX(), 
+      tag_pose.getOrigin().getY(), 
+      tag_pose.getOrigin().getZ()
+    )
+  );
+  ROS_DEBUG_STREAM_THROTTLE(1.0, "Frame " << camera_model_.cameraInfo().header.frame_id << " has target at pixel coordinate x: " << image_pt.x << ", y: " << image_pt.y);
+
+  // check if the image is within the target fov (from image width and height in pixels), return true if it is and false otherwise
+  return fov_pixel_buffer_width_ <= image_pt.x && image_pt.x < (camera_model_.cameraInfo().width - fov_pixel_buffer_width_) && fov_pixel_buffer_height_ <= image_pt.y && image_pt.y < (camera_model_.cameraInfo().height - fov_pixel_buffer_height_);
+}
+
+bool ContinuousDetector::isTagInDetectionRange(const tf2::Transform& tag_pose) const
+{
+  const auto curr_dist2 = tag_pose.getOrigin().length2();
+  ROS_DEBUG_STREAM_THROTTLE(1.0, "Frame " << camera_model_.cameraInfo().header.frame_id << " has dist2 to targ: " << curr_dist2 << ", however min is " << min_detection_dist2_ << " and max dist2 is " << max_detection_dist2_);
+  return min_detection_dist2_ <= curr_dist2 && curr_dist2 <= max_detection_dist2_;
+}
+
+void ContinuousDetector::targetPoseCallback(const geometry_msgs::PoseStamped& pose_msg)
+{
+  tf2::fromMsg(pose_msg.pose, tag_pose_);
+  // the target pose should be from the base frame
+  camera_frame_transformer_.setBaseFrame(pose_msg.header.frame_id);
+  pose_base_frame_header_ = pose_msg.header;
```

Comment:
I may remove this.. I was using this for the time stamp but honestly, I can probably just use the image_rect->header rather than taking this one